refactor(hardware-test): improve devenv output

HerrCooker opened the pull request:
- removed unnecessary information
- removed unnecessary lines
- introduced formatted summed up list of all test results
- introduced color coding to improve readability

Example Output:
<img width="1448" height="591" alt="Screenshot from 2025-07-10 16-09-54" src="https://github.com/user-attachments/assets/d1ff37e4-9734-4796-b4ba-91a727739383" />

DavidFederl commented on devenv.nix:
```diff
@@ -151,16 +158,44 @@ in
       package = pkgs.bash;
       description = "apply clang-format to src,test directory";
```
Please update the description!

HerrCooker commented:
Additional Changes: 
- Before, the entire `run_test` script was silenced. So if you wanted to just run one test, there was no output (OK or FAILED)
- Now the `run_test` script is only silenced when its being called from the `run_tests` script (where we dont want to see each OK or FAILED)
- Changed the description of  `run_test`
- Added color coding to `run_test`
- removed `#!/usr/bin/env bash`

DavidFederl reviewed (approved):
👌
